shared/common-adapters/badge.native.js

```diff
@@ -48,4 +48,49 @@ const styles = styleSheetCreate({
   },
 })
 
-export default Badge
+export class Badge2 extends React.Component<Badge2Props> {
+  static defaultProps = {fontSize: 11, radius: 12, leftRightPadding: 6, topBottomPadding: 4}
+
+  render() {
+    return (
+      <Box
+        style={collapseStyles([
+          badge2Styles.badge,
+          {
+            borderRadius: this.props.radius,
+            height: this.props.radius * 2,
+            minWidth: this.props.radius * 2,
+            paddingBottom: this.props.topBottomPadding,
+            paddingLeft: this.props.leftRightPadding,
+            paddingRight: this.props.leftRightPadding,
+            paddingTop: this.props.topBottomPadding,
+          },
+          this.props.badgeStyle,
+        ])}
+      >
+        <Text
+          type="BodyTinySemibold"
+          style={collapseStyles([
+            badge2Styles.text,
+            this.props.badgeNumberStyle,
+            {fontSize: this.props.fontSize, lineHeight: this.props.fontSize + 5},
```

Comment:
what is this 5 about

Reply:
In all of our text styles, [lineHeight is exactly + 5 of font size](https://github.com/keybase/client/blob/nathan/DESKTOP-7129-additional-badge-fixes/shared/common-adapters/text.meta.native.js#L25). I needed a way to scale line height with font size and that was the first thing I came up with – although now that I'm looking at it again, I don't like how hacky it is. Perhaps a better solution would be to extract the `fontSizeToSizeStyle` function into a utility file and call it here?

Reply:
does just aligning it center just fix this and you can ignore lineheight?

Reply:
It doesn't by default but setting `lineHeight` equal to the `height` [does work](https://stackoverflow.com/questions/8865458/how-do-i-vertically-center-text-with-css)!

A problem with this approach, however, is that we can't seem add any top or bottom padding – this seems to throw off the centering. I don't think that's supper important considering the top and bottom paddings are sort of implicitly determined by height, though.